client/src/client-only-routes/ShowProjectLinks.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable react/jsx-sort-props */
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import '../components/layouts/project-links.css';
+import { Button, Modal } from '@freecodecamp/react-bootstrap';
+import { maybeUrlRE } from '../utils';
+import { Spacer } from '../components/helpers';
+import { projectMap } from '../resources/certAndProjectMap';
+import { Link } from 'gatsby';
+import SolutionViewer from '../components/settings/SolutionViewer';
+import { find } from 'lodash';
+
+const propTypes = {
+  certName: PropTypes.string,
+  name: PropTypes.string,
+  user: PropTypes.shape({
+    completedChallenges: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string,
+        solution: PropTypes.string,
+        githubLink: PropTypes.string,
+        files: PropTypes.array
+      })
+    ),
+    username: PropTypes.string
+  }),
+  username: PropTypes.string
+};
+
+const initSolutionState = {
```

Comment:
Could we split this up?  The React docs [recommend multiple](https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables) `useState`s, since state changes are not merged by default.

Reply:
This is what the React Docs say:
> However, we recommend to split state into multiple state variables **based on which values tend to change together**.

Seeing as the only place the state is changed is:
```javascript
const onClickHandler = () =>
      setSolutionState({
        projectTitle,
        files,
        solution,
        isOpen: true
      });
```
So, they all change together. I do not see any benefit from splitting this object up.

Reply:
Ah, that's right.  Thanks for checking this carefully - your way seems best.